packages/microsoft_dnsserver/data_stream/analytical/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,729 @@
+        }
+
+  - script:
+      lang: painless
+      description: This script processor rename the fields under the analytical object.
+      if: ctx.dnsserver?.analytical != null
+      tag: painless_to_rename_fields_under_dnsserver_analytical
+      params:
+        "AdditionalInfo": "additional_info"
+        "BufferSize": "bytes_sent"
+        "CacheScope": "cache_scope"
+        "DNSSEC": "dnssec"
+        "ElapsedTime": "elapsed_time"
+        "GUID": "guid"
+        "PacketData": "packet_data"
+        "PolicyName": "policy_name"
+        "QNAME": "question_name"
+        "QueriesAttached": "queries_attached"
+        "QXID": "qxid"
+        "Reason": "reason"
+        "RecursionDepth": "recursion_depth"
+        "RecursionScope": "recursion_scope"
+        "Scope": "scope"
+        "Secure": "secure"
+        "XID": "xid"
+        "Zone": "zone"
+        "ZoneScope": "zone_scope"
+      source: |
+        def renameKeys(Map src, Map keyMap) {
+          def dst = new HashMap();
+          for (def entry: src.entrySet()) {
+            def key = entry.getKey();
+            def value = entry.getValue();
+            if (value instanceof Map) {
+              if (keyMap.containsKey(key)) {
+                dst[keyMap[key]] = renameKeys(value, keyMap);
+              } else {
+                dst[key] = renameKeys(value, keyMap);
+              }
+            } else if (value instanceof List) {
+              def updatedList = [];
+              for (def item: value) {
+                if (item instanceof Map) {
+                  updatedList.add(renameKeys(item, keyMap));
+                } else {
+                  updatedList.add(item);
+                }
+              }
+              if (keyMap.containsKey(key)) {
+                dst[keyMap[key]] = updatedList;
+              } else {
+                dst[key] = value;
+              }
+            } else {
+              if (keyMap.containsKey(key)) {
+                dst[keyMap[key]] = value;
+              } else {
+                dst[key] = value;
+              }
+            }
+          }
+          return dst;
+        }
+        ctx.dnsserver.analytical = renameKeys(ctx.dnsserver.analytical, params)
+
+# ECS fields
+  - set:
+      field: dns.id
+      copy_from: dnsserver.analytical.xid
+      ignore_empty_value: true
+      tag: set_dns_id
+  - set:
+      field: dns.question.name
+      copy_from: dnsserver.analytical.question_name
+      ignore_empty_value: true
+      tag: set_dns_question_name
+  - set:
+      field: dns.question.type
+      copy_from: dnsserver.analytical.question_type
+      ignore_empty_value: true
+      tag: set_dns_question_type
+  - set:
+      field: dns.response_code
+      copy_from: dnsserver.analytical.response_code
+      ignore_empty_value: true
+      tag: set_dns_question_response
+  - set: 
+      field: event.outcome
+      value: success
+      if: ctx.dns?.response_code != null && ctx.dns.response_code == "NoError"
```

```suggestion
      if: ctx.dns?.response_code == "NoError"
```
(similar for the `ctx.dns?.response_code != null && ctx.dns.response_code == VALUE` checks below since VALUE != null)